clients/src/main/java/org/apache/kafka/common/requests/JoinGroupRequest.java

```diff
@@ -120,10 +122,10 @@ public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
         return new JoinGroupResponse(new JoinGroupResponseData()
                 .setThrottleTimeMs(throttleTimeMs)
                 .setErrorCode(Errors.forException(e).code())
-                .setGenerationId(JoinGroupResponse.UNKNOWN_GENERATION_ID)
-                .setProtocolName(JoinGroupResponse.UNKNOWN_PROTOCOL)
-                .setLeader(JoinGroupResponse.UNKNOWN_MEMBER_ID)
-                .setMemberId(JoinGroupResponse.UNKNOWN_MEMBER_ID)
+                .setGenerationId(UNKNOWN_GENERATION_ID)
```

Comment:
nit: I think it would be a little cleaner to rely on the defaults from the schema. So if we don't have a `GroupMetadata` object, then we don't set anything.

Reply:
Sorry, wrong request type here. Please ignore.